src/components/Toolbar.tsx

```diff
@@ -132,15 +133,19 @@ const Toolbar: FC<ToolbarProps> = ({ customize, onSelect, selected }) => {
       setRightArrowIsShown(el.offsetWidth + el.scrollLeft < el.scrollWidth - 20)
     }
   }, [])
-
   /** Handles toolbar scroll event. */
   const onScroll = useCallback(
     (e: React.UIEvent<HTMLElement>) => {
+      let lastScrollY = 0
```

Comment:
One issue here is that `lastScrollY` is always reset to 0 whenever this callback gets called.

Here is one suggested approach that seems trigger the haptic feedback for every ~50 pixels scrolled. You might want to use that as a starting point: https://github.com/cybersemics/em/commit/c0dabee32ff736d7f45b1e0492dfa4e2a7ecb7c6